Sources/SwiftFormatRules/AlwaysUseLowerCamelCase.swift

```diff
@@ -19,31 +19,72 @@ import SwiftSyntax
 /// Lint: If an identifier contains underscores or begins with a capital letter, a lint error is
 ///       raised.
 public final class AlwaysUseLowerCamelCase: SyntaxLintRule {
+  /// Stores function decls that are test cases.
+  private var testCaseFuncs = Set<FunctionDeclSyntax>()
+
+  public override func visit(_ node: SourceFileSyntax) -> SyntaxVisitorContinueKind {
+    // Tracks whether "XCTest" is imported in the source file before processing individual nodes.
+    setImportsXCTest(context: context, sourceFile: node)
+    return .visitChildren
+  }
+
+  public override func visit(_ node: ClassDeclSyntax) -> SyntaxVisitorContinueKind {
+    // Check if this class is an `XCTestCase`, otherwise it cannot contain any test cases.
+    guard context.importsXCTest == .importsXCTest else { return .visitChildren }
+    guard let inheritanceClause = node.inheritanceClause else { return .visitChildren }
+    let inheritsXCTestCase = inheritanceClause.inheritedTypeCollection.contains(where: {
+      guard let typeIdentifier = $0.typeName.as(SimpleTypeIdentifierSyntax.self) else {
+        return false
+      }
+      return typeIdentifier.name.text == "XCTestCase"
+    })
+    guard inheritsXCTestCase else { return .visitChildren }
+
+    // Identify and store all of the function decls that are test cases.
+    let testCases = node.members.members.compactMap {
+      $0.decl.as(FunctionDeclSyntax.self)
+    }.filter {
+      // Filter out non-test methods using the same heuristics as XCTest to identify tests.
+      // Test methods are methods that start with "test", have no arguments, and void return type.
+      $0.identifier.text.starts(with: "test")
+        && $0.signature.input.parameterList.isEmpty
+        && $0.signature.output == nil
```

Comment:
In recent versions of XCTest, test methods are allowed to be declared `throws`, so we need to handle that possibility as well.

Reply:
The `throws` keyword will be alright, and I added a test for it. But this did make me realize I wasn't handling `-> Void` or `-> ()` correctly.